util/headerreader/header_reader.go

```diff
@@ -201,6 +213,22 @@ func (s *HeaderReader) possiblyBroadcast(h *types.Header) {
 	}
 }
 
+func (s *HeaderReader) getPendingCallBlockNumber() (*big.Int, error) {
+	if s.isParentChainArbitrum {
+		arbSys, err := precompilesgen.NewArbSys(types.ArbSysAddress, s.client)
```

Comment:
Let's make this a field of `HeaderReader` so we don't need to create it each time (I think it involves parsing the ABI JSON which might be a bit expensive).

Reply:
done.